Tribes/src/app/building.service.ts

```diff
@@ -0,0 +1,20 @@
+import { BUILDINGS } from './mock-building';
+import { Building } from './building';
+import { Injectable } from '@angular/core';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BuildingService {
+
+  constructor() { }
+
+  getBuildings(): Building[] {
+    return BUILDINGS;
+  }
+
+  getBuilding(type: string) {
+    return BUILDINGS.find(building => building.type === type);
```

Comment:
if we want to find a specific building, we should find it by `id` or something, because i think this is returning the first building with that type

Reply:
I wrote a seperate method to find buildings by ID

Reply:
then what is this for?

Reply:
To list the icons on the dashboard and to use it in the routerlink to direct it to the detailed page of each building type. I use this method in the building-type.component.ts

Reply:
Please remove this if not used.